src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DefaultBoundAttributeDescriptorBuilder.cs

```diff
@@ -94,26 +87,19 @@ public override void BindAttributeParameter(Action<BoundAttributeParameterDescri
 
     public BoundAttributeDescriptor Build()
     {
-        var diagnostics = Validate();
-        if (_diagnostics != null)
-        {
-            diagnostics ??= new();
-            diagnostics.UnionWith(_diagnostics);
-        }
+        using var _ = HashSetPool<RazorDiagnostic>.GetPooledObject(out var diagnostics);
 
-        var parameters = Array.Empty<BoundAttributeParameterDescriptor>();
-        if (_attributeParameterBuilders != null)
-        {
-            // Attribute parameters are case-sensitive.
-            var parameterset = new HashSet<BoundAttributeParameterDescriptor>(BoundAttributeParameterDescriptorComparer.Default);
-            for (var i = 0; i < _attributeParameterBuilders.Count; i++)
-            {
-                parameterset.Add(_attributeParameterBuilders[i].Build());
-            }
+        Validate(diagnostics);
 
-            parameters = parameterset.ToArray();
+        if (_diagnostics is { } existingDiagnostics)
```

Comment:
```suggestion
        if (_diagnostics is not null existingDiagnostics)
```

Reply:
I think I've seen that language proposal, but I think you're jumping the gun by a few versions :)

Reply:
😅 this is what I get for doing a quick review on my way out the door. Should just be 

```csharp
if (_diagnostics is not null)
```

Reply:
Then I'd have to read the `_diagnostics` field a second time, which I was trying to avoid.

Reply:
FWIW, I'm changing this such that there won't even be pattern matching here.